--- libmodman/CMakeLists.txt
+++ libmodman/CMakeLists.txt
@@ -9,7 +9,7 @@
 else(WIN32)
   set(CMAKE_CXX_FLAGS "-fvisibility=hidden ${CMAKE_CXX_FLAGS}")
 endif(WIN32)
-include_directories(${CMAKE_SOURCE_DIR})
+include_directories(${PROJECT_SOURCE_DIR})
 
 # PkgConfig file
 if(NOT WIN32 AND NOT APPLE)
@@ -42,7 +42,7 @@
 endif()
 
 # Define the library itself
-add_library(modman SHARED
+add_library(modman
             module.hpp
             module_manager.hpp
             module_manager.cpp)
@@ -54,4 +54,4 @@
 install(FILES   module_manager.hpp module.hpp DESTINATION ${INCLUDE_INSTALL_DIR}/${PROJECT_NAME})
 
 ### Tests
-add_testdirectory(test)
+#add_testdirectory(test)
