--- src/base/main/mainReal.c
+++ src/base/main/mainReal.c
@@ -83,7 +83,7 @@
   SeeAlso     []
 
 ***********************************************************************/
-int Abc_RealMain( int argc, char * argv[] )
+ABC_DLL int Abc_RealMain( int argc, char * argv[] )
 {
     Abc_Frame_t * pAbc;
     Vec_Str_t* sCommandUsr = Vec_StrAlloc(1000);
--- src/base/main/main.c
+++ src/base/main/main.c
@@ -2,7 +2,7 @@
 
 ABC_NAMESPACE_IMPL_START
 
-int Abc_RealMain(int argc, char *argv[]);
+int Abc_RealMain(int argc, char *argv[]);
 
 ABC_NAMESPACE_IMPL_END
 
