--- lib/configure
+++ lib/configure
@@ -143,7 +143,7 @@
 fi

 echo > pngquant-gcccheck.c "int main(){}"
-if ! "$CC" -std=c99 -o pngquant-gcccheck pngquant-gcccheck.c; then
+if ! $CC -std=c99 -o pngquant-gcccheck pngquant-gcccheck.c; then
     rm -f pngquant-gcccheck pngquant-gcccheck.c
     error "Compiler" "$CC failed to compile anything (make sure it's installed and supports C99)"
 fi
@@ -206,7 +206,7 @@
     fi
 else
     # silence warnings about omp pragmas
-    cflags "-Wno-unknown-pragmas"
+    # cflags "-Wno-unknown-pragmas"
     conditional_cflags "-wd3180" # ICC
     status "OpenMP" "no"
 fi
--- /dev/null
+++ lib/dllmain.c
@@ -0,0 +0,7 @@
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
+BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+{
+    return 1;
+}
--- lib/Makefile
+++ lib/Makefile
@@ -45,8 +45,8 @@
 java-dll:
 	$(MAKE) CFLAGS="$(CFLAGS) -DIMAGEQUANT_EXPORTS" $(JNIDLL)

-$(DLL) $(DLLIMP): $(OBJS)
-	$(CC) -fPIC -shared -o $(DLL) $^ $(LDFLAGS) -Wl,--out-implib,$(DLLIMP),--output-def,$(DLLDEF)
+$(DLL) $(DLLIMP): $(OBJS) dllmain.o
+	$(LD) -DLL /OUT:$(DLL) $^ /IMPLIB:$(DLLIMP)

 $(STATICLIB): $(OBJS)
 	$(AR) $(ARFLAGS) $@ $^

