--- perplex.cpp
+++ perplex.cpp
@@ -100,6 +100,41 @@
 "  -v, --version\t\tprint perplex version number and exit\n"
 ;
 
+#ifdef _WIN32
+# define WIN32_LEAN_AND_MEAN
+# include <windows.h>
+#else
+# include <libgen.h>
+# include <unistd.h>
+#endif
+#include <cstring>
+
+#define DEFAULT_PATH "."
+#define DEFAULT_TEMPLATE_FN "perplex_template.c"
+
+const char *get_default_template_path() {
+    static char static_path[8192];
+#ifdef _WIN32
+    char joiner = '\\';
+    int result = GetModuleFileNameA(NULL, static_path, sizeof(static_path));
+    if (result != ERROR_SUCCESS) {
+        memcpy(static_path, DEFAULT_PATH, strlen(DEFAULT_PATH));
+    }
+#else
+    char joiner = '/';
+    ssize_t nb = readlink("/proc/self/exe", static_path, sizeof(static_path));
+    if (nb != -1) {
+        static_path[nb] = '\0';
+        dirname(static_path);
+    } else {
+        memcpy(static_path, DEFAULT_PATH, strlen(DEFAULT_PATH));
+    }
+#endif
+    strncat(static_path, &joiner, 1);
+    strncat(static_path, DEFAULT_TEMPLATE_FN, sizeof(DEFAULT_TEMPLATE_FN));
+    return static_path;
+}
+
 int main(int argc, char *argv[])
 {
     int c;
@@ -109,7 +144,7 @@
     void *parser;
     perplex_t scanner;
     appData_t *appData;
-    char defaultTemplate[] = "perplex_template.c";
+    const char *defaultTemplate = get_default_template_path();
     FILE *inFile;
     FILE *outFile = stdout;
     FILE *templateFile = NULL;
